pkg/collector/reconcile/horizontalpodautoscaler.go

```diff
@@ -87,6 +86,15 @@ func expectedHorizontalPodAutoscalers(ctx context.Context, params Params, expect
 		updated.SetOwnerReferences(desired.GetOwnerReferences())
 		setAutoscalerSpec(params, autoscalingVersion, updated)
 
+		// update metrics separately because the structure of the metrics can change.
+		if autoscalingVersion == autodetect.AutoscalingVersionV2Beta2 {
+			desiredMetrics := desired.(*autoscalingv2beta2.HorizontalPodAutoscaler).Spec.Metrics
+			updated.(*autoscalingv2beta2.HorizontalPodAutoscaler).Spec.Metrics = desiredMetrics
+		} else { // autoscalingv2
+			desiredMetrics := desired.(*autoscalingv2.HorizontalPodAutoscaler).Spec.Metrics
+			updated.(*autoscalingv2.HorizontalPodAutoscaler).Spec.Metrics = desiredMetrics
+		}
+
```

Comment:
Should this be part of the `setAutoscalerSpec` since we're manipulating the spec?

Reply:
Thanks for the feedback! I've now placed this logic in `setAutoscalerSpec`